src/lexers/jsx-lexer.js

```diff
@@ -54,16 +54,21 @@ export default class JsxLexer extends JavascriptLexer {
       const entry = {}
       entry.key = getKey(tagNode)
 
-      const defaultValue = this.nodeToString.call(this, node, sourceText)
+      if(!entry.key) {
+        entry.key = this.nodeToString.call(this, node, sourceText);
+      } else {
```

Comment:
It seems that when there is no key, default value is never set.
When `useKeyAsDefaultValue` is true, it should still set default value though

Reply:
Ok, you're perfectly right. If `useKeyAsDefaultValue` is true, the defaultValue should be set the same as the key. I thought this check was done later  and not in the lexer. How do I access the options? Is in the options passed to the lexer constructor?

Reply:
You never checked the option before my changes. What was the previous behavior? You always set the default value the same as the key, regardless of the `useKeyAsDefaultValue` option, which is what I'm complaining about in the issue. Was it intended?

Reply:
Looking at the code, it seems that the main options, included `useKeyAsDefaultValue`, are not passed to the lexers. The option seems to be used in the transformer, after the file has been parsed by the lexer. So I guess no other test case should be added to the lexers because the fallback of the value to the key is handled elsewhere

Reply:
Any news on this?

Reply:
I'm totally fine that you pass `options` as a second argument [here](https://github.com/i18next/i18next-parser/blob/master/src/parser.js#L69). That way it will be easy to access this and do the check in the lexer.

Reply:
Ok, but why? Lexers already works without knowing about those options and the check is done later. So basically your code is working without passing the options to the lexers and I don't see any reason to start doing it now.

Reply:
Hmm, I agree with your reasoning, it's probably not up to the lexer to decide what default value should be - it's supposed to be responsible only for making sense of the code. It's likely that the flag was never working as intended, not sure. Probably the logic to assign the default from key based on the flag should be in parser then.

Reply:
However, I'd interpret this in a way that is contrary to yours, if i18nkey is not set, I assume no key is set, and children is always the default value, but in the parser clean up the key to be the default value if missing (not sure how it impacts other parts, and pretty sure that right now key is being set to default value in lexer anyway)

Reply:
I think there's not a single way to view this, it's just a matter of opinions. From [your comment](https://github.com/i18next/i18next-parser/issues/206#issuecomment-589888363) on the issue it seems like also react-i18next shares your opinion. I still prefer my way, after all there's an option in i18next to fallback to the key if the value is missing which would correctly handle this case as well. Feel free to not continue to review this PR and close it and thanks for all of the attention :)

Reply:
I'm not sure about your exact use case, but I don't think in general that lexer is working incorrectly right now. I assume you want to skip default values altogether when parsing and have them all empty? Maybe just an additional config flag that disables writing default values in parser/transform level is enough?

Reply:
I think there's such an option already, it's `useKeysAsDefaultValue`. It is just not considered when parsing the `<T>` component because the only value present, i.e. `<T>A text</T>` is considered both the key and the value. I'm just saying that "A text" should be considered only as the key, not the value.

> only maybe it shouldn't make assumptions about key from default value

In my opinion it is doing assumptions about the default value from the key.

Again, I really think the problem is how the children of the T component is considered when the attribute `i18nKey` is not present: if the children are considered being the key, the implementation is now wrong. When the children are considered both the default value and the key, the current implementation is right

Reply:
Something else, reading the documentation of react-i18next, it seems to suggest that, when the `i18nKey` is missing, the value (the children) must considered as the key. It doesn't say the value should be considered as the value. You can find this description [here](https://react.i18next.com/latest/trans-component#trans-props)

Reply:
if key is missing, the children go through a function (defined in config) to become a key, by default I think it's just the same as default value:
```
hashTransKey: function(defaultValue) {
      // return a key based on defaultValue or if you prefer to just remind you should set a key return false and throw an error
    }
```
And I'm fine with key being the same as default value in case it's missing for Trans component, but I don't think the actual default value should then be omitted based on `useKeysAsDefaultValue` flag alone.

Reply:
Ok, it's fine. So let's close this MR in case. I cannot find the `hashTransKey` in config, can you point me to its description?

Reply:
Ok, found it (https://react.i18next.com/legacy-v9/trans-component#additional-options-on-i-18-next-init).Yes, it seems to suggest the value inside the Trans component is the default value

Reply:
The parser doesn't have this config value though, maybe it makes sense to add it though as it would solve your issue